packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts

```diff
@@ -34,7 +35,57 @@ export class LazyListStackResources implements ListStackResources {
   }
 }
 
-export class CfnEvaluationException extends Error {}
+export interface LookupExport {
+  lookupExport(name: string): Promise<AWS.CloudFormation.Export | undefined>;
+}
+
+export class LookupExportError extends Error { }
+
+export class LazyLookupExport implements LookupExport {
+  private cachedExports: { [name: string]: AWS.CloudFormation.Export } = {}
+
+  constructor(private readonly sdk: ISDK) { }
+
+  async lookupExport(name: string): Promise<AWS.CloudFormation.Export | undefined> {
+    if (this.cachedExports[name]) {
+      return this.cachedExports[name];
+    }
+
+    for await (const cfnExport of this.listExports()) {
+      if (!cfnExport.Name) {
+        throw new LookupExportError(`Unable to handle CloudFormation Export without Name! ${JSON.stringify(cfnExport)}`);
+      }
+      this.cachedExports[cfnExport.Name] = cfnExport;
+
+      if (cfnExport.Name === name) {
+        return cfnExport;
+      }
```

Comment:
I wanted to provide an implementation that performed well under scenarios where a user had many many exports so I wanted to avoid caching all exports up front. But perhaps I can update my implementation to retain the iterator and pick up searching where it left off in subsequent calls

Reply:
What would be the benefit of listing exports for each lookup for a hotswap? Do you expect the list to be changed in each lookup?

Reply:
Would not [this](https://github.com/aws/aws-cdk/pull/27292/files#diff-ff9724092d7f1e6bf3141e51ddb00ecbc2148e16f141b0bd2074d6edc60c16b6R69) list all the exports present?

Reply:
The [linked function](https://github.com/aws/aws-cdk/pull/27292/files#diff-ff9724092d7f1e6bf3141e51ddb00ecbc2148e16f141b0bd2074d6edc60c16b6R69) is a generator function so it will yield pages as requested by the [loop here](https://github.com/aws/aws-cdk/pull/27292/files#diff-ff9724092d7f1e6bf3141e51ddb00ecbc2148e16f141b0bd2074d6edc60c16b6R54) that is searching for the export

The benefit isn't so much about ensuring we have the latest, but more about avoiding cost of iterating all the exports if we don't need to (i.e. if found in the first page then no need to enumerate the rest)